### PR TITLE
fix: patch correctly config in `talosctl upgrade-k8s`

### PR DESCRIPTION
**Pull request by smira**
The current code was stipping non-`v1alpha1.Config` documents. Provide a proper method in the config provider, and update places using it.

### CODE REVIEW

File: pkg/machinery/config/provider.go [hunk 1]
```diff
@@ -42,6 +42,9 @@ type Provider interface {
 	// Clone returns a copy of the Provider.
 	Clone() Provider
 
+	// PatchV1Alpha1 the container's v1alpha1.Config while preserving other config documents.
```
Comment (utkuozdemir) [inline pkg/machinery/config/provider.go:45]:
```suggestion
	// PatchV1Alpha1 patches the container's v1alpha1.Config while preserving other config documents.
```

**Review by utkuozdemir — APPROVED** [review]

File: pkg/machinery/config/provider.go (hunk 1 above)
Comment (smira) [reply pkg/machinery/config/provider.go]: thanks, fixed

**Review by frezbo — APPROVED** [review]

**Review by Unix4ever — APPROVED** [review]

**Pull request merged**